components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,358 @@
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.patientResource = (DefaultPatientResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+
+
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
+
+        doReturn(this.patient).when(this.repository).getPatientById(this.id);
+        doReturn(null).when(this.patient).getDocument();
+
+        this.uri = new URI(this.uriString);
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+
+        ReflectionUtils.setFieldValue(this.patientResource, "uriInfo", this.uriInfo);
+    }
+
+    //----------------------------Get Patient Tests----------------------------
+
+    @Test
+    public void getPatientCantFindPatient() throws XWikiRestException {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("No such patient record: [{}]", this.id);
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientUserDoesNotHaveAccess() throws XWikiRestException {
+        doReturn(false).when(this.access).hasAccess(Right.VIEW, null, null);
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("View access denied to user [{}] on patient record [{}]", this.currentUser, this.id);
+        Assert.assertEquals(Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void getPatientAddsLinksToPatientJSON() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, null, null);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        Assert.assertTrue(response.getEntity() instanceof JSONObject);
+
+        Map<String, Map<String, String>> json = (Map)response.getEntity();
+        Assert.assertThat(json, hasValue(hasEntry("rel", Relations.SELF)));
+        Assert.assertThat(json, hasValue(hasEntry("href", this.uriString)));
+        Map<String, List<MediaType>> actualMap = (Map)response.getMetadata();
+        Assert.assertThat(actualMap, hasValue(hasItem(MediaType.APPLICATION_JSON_TYPE)));
+    }
+
+    @Test
+    public void getPatientAlwaysSendsLoggerMessageOnRequest() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, null, null);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
```

Comment:
This is a small point, but to reduce the number of lines this has to go through you could just make this.repository.getPatientById(id) return null and the call would end faster with the log you are checking still happening